### PR TITLE
Replace black/white list terms with more descriptive language

### PR DESCRIPTION
**Pull request by melix**
There are always better words to choose from than connotated
language which can be either offensive or associating one
people's color with something bad.

This commit is also a first step in acting, following our
company's statement:

As a distributed company that calls many countries home, we have
been shocked and saddened by the death of George Floyd and the police
response to the protests.
People of color are being murdered and brutalized by police with few
consequences. Gradle stands in solidarity  with protesters demanding
accountability.

We urge governments worldwide, including the US federal, state, and
local governments to act swiftly and meaningfully to address the root
cause of these protests and take immediate measures to stop unlawful
killings by police.

Gradle folks across the company have donated thousands of dollars to
organizations working to address racism and systemic injustices in the
world and Gradle will be evaluating within our own organization how to
do more.

If you would like to do something, please consider:

   - Donating money to Amnesty International, Black Visions Collective,
NAACP LDF, or National Bail Fund.
   - Volunteering time with organizations in your local community that
fight for equality and justice
   - Writing to your government representatives

Fixes #?

### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [ ] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [ ] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by melix** [comment]
@bot-gradle test this

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java [hunk 1]
```diff
@@ -33,11 +33,11 @@
 import org.gradle.api.internal.artifacts.ivyservice.ArtifactCacheLockingManager;
 import org.gradle.api.internal.artifacts.ivyservice.ArtifactCacheMetadata;
 import org.gradle.api.internal.artifacts.ivyservice.ArtifactCachesProvider;
-import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ConnectionFailureRepositoryBlacklister;
+import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ConnectionFailureRepositoryDisabler;
 import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ChangingValueDependencyResolutionListener;
 import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ModuleDescriptorHashCodec;
 import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ModuleDescriptorHashModuleSource;
-import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryBlacklister;
+import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryDisabler;
```
Comment (lptr) [inline subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java:40]: Should this be a `RepositoryFilter`?

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy [hunk 2]
```diff
@@ -140,7 +140,7 @@ class DependencyUnresolvedModuleIntegrationTest extends AbstractHttpDependencyRe
         protocol << ['http', 'https']
     }
 
-    def "blacklists repository from later resolution within the same build on HTTP timeout "() {
+    def "disables repository from later resolution within the same build on HTTP timeout "() {
```
Comment (lptr) [inline subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy:143]: How about:

```suggestion
    def "prevents using repository in later resolution within the same build on HTTP timeout"() {
```

(What's the trailing space do in the method name?)

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java (hunk 1 above)
Comment (melix) [reply subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java:40]: No, repository filtering is a [very different concept](https://docs.gradle.org/6.5.1/userguide/declaring_repositories.html#sec:repository-content-filtering) so i would be confusing. Here it's not even public API, it's an internal thing which is stateful and disabled repositories during resolution.

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy (hunk 2 above)
Comment (melix) [reply subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy:143]: +1

**Comment by melix** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/resolution_rules.adoc [hunk 3]
```diff
@@ -37,10 +37,10 @@ include::sample[dir="snippets/dependencyManagement/customizingResolution-resolut
 include::sample[dir="snippets/dependencyManagement/customizingResolution-resolutionStrategy/kotlin",files="build.gradle.kts[tags=custom-versioning-scheme]"]
 ====
 
-[[sec:blacklisting_version]]
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/resolution_rules.adoc:40]: Should we go the extra mile and see if we can keep incoming links to that ID end up resolving to the new one?

Comment (melix) [reply subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/resolution_rules.adoc:40]: No idea how to do this, though.

**Pull request merged**